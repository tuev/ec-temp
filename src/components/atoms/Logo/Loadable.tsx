/**
 *
 * Asynchronously loads the component for Logo
 *
 */

import loadable from 'utils/loadable'

export default loadable(() => import('./index'))
