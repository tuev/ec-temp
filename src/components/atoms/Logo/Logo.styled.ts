/**
 *
 * Styled for Logo
 *
 */

import styled from 'styled-components'
import { prop } from 'styled-tools'

export const LogoImg = styled.div<{ img: string; size?: string | number }>`
  background-image: url(${prop('img')});
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  width: ${prop('size', '40px')};
  height: ${prop('size', '40px')};
`
