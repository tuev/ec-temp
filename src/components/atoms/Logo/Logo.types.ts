export type LogoProp = {
  size?: number | string
}
