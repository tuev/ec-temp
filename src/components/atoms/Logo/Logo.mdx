# Stories mdx for Logo

import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './Logo.stories.tsx'
import Logo from './Logo.tsx'

<Story name="basic" />