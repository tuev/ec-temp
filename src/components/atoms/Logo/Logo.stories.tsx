/**
 *
 *  Stories for Logo
 *
 */

import React from 'react'
import { action } from '@storybook/addon-actions'
import Logo from './Logo'

export default {
  component: Logo,
  includeStories: ['basic'],
  title: 'Logo',
}

export const basic = () => (
  <div>
    <Logo size="72px" />
  </div>
)
