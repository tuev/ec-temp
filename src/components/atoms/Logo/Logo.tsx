/**
 *
 * Logo
 *
 */

import React, { FC } from 'react'
import logoImg from './img/logo_transparent.png'
import { LogoImg } from './Logo.styled'
import { LogoProp } from './Logo.types'

const Logo: FC<LogoProp> = (props: LogoProp) => {
  return <LogoImg {...props} img={logoImg} />
}

export default Logo
