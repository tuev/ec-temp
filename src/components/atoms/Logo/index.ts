/**
 *
 * Index for Logo
 *
 */

import Logo from './Logo'

export default Logo
