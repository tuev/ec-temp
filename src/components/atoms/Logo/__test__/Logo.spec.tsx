/**
 *
 * Tests for Logo
 *
 */

import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import Logo from '..'

describe('App Logo', () => {
  it('Test get default Logo no params ', () => {
    // test snapshot
    const wrapper = render(<Logo />)
    expect(wrapper).toMatchSnapshot()
  })
})
