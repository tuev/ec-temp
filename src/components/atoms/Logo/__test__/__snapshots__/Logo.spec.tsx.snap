// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Logo Test get default Logo no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  background-image: url(logo_transparent.png);
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  width: 40px;
  height: 40px;
}

<body>
    <div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm ihbyqD"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
